Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizerError.swift

```diff
@@ -1,9 +1,65 @@
-public class DataSynchronizerError: Error, CustomStringConvertible {
-    public var description: String
+/// A set of errors that can occur during the synchronization process.
+public enum DataSynchronizerError: Error, CustomStringConvertible {
+    /**
+        Any error related to decoding a document
+        This will likely be due to a malformed sync version
+        or an unsupported change event.
+    */
+    case decodingError(_ message: String)
+    /**
+        An unsupported sync protocol version has been
+        supplied with a document. The error message
+        contains the faulty version.
+    */
+    case unsupportedProtocolVersion(_ message: String)
```

Comment:
isn't this non-recoverable?

Reply:
No, a non-recoverable error would be one that forces the data synchronizer to shut down, e.g., not being able to interact with embedded mongo (for whatever reason).

An unsupported protocol version would, however, "pause" the document that is being sync'd. This would then be made available in the `pausedIds` field of the user-exposed `Sync` class. Users could then choose how to fix the error.